packages/sdk-react/src/components/payables/PayablesTable/hooks/usePaymentHandler.tsx

```diff
@@ -9,49 +9,135 @@ import { useLingui } from '@lingui/react';
 
+  const { data: paymentMethods } =
+    api.entities.getEntitiesIdPaymentMethods.useQuery({
+      path: { entity_id: monite.entityId },
+    });
+
   const paymentIntentQuery = api.paymentIntents.getPaymentIntents.useQuery({
     query: { object_id: payableId },
   });
 
+  const createPaymentLinkMutation =
+    api.paymentLinks.postPaymentLinks.useMutation({});
+
   const paymentLinkId = paymentIntentQuery.data?.data?.[0]?.payment_link_id;
 
   const paymentLinkQuery = api.paymentLinks.getPaymentLinksId.useQuery(
     { path: { payment_link_id: paymentLinkId || '' } },
     { enabled: !!paymentLinkId }
   );
 
-  const handleCloseModal = () => {
+  const payMutation = api.payables.postPayablesIdMarkAsPaid.useMutation(
+    undefined,
+    {
+      onSuccess: (payable) =>
+        Promise.all([
+          api.payables.getPayablesId.invalidateQueries(
+            { parameters: { path: { payable_id: payable.id } } },
+            queryClient
+          ),
+          api.payables.getPayables.invalidateQueries(queryClient),
+        ]),
+      onError: (error) => {
+        toast.error(error.toString());
+      },
+    }
+  );
+
+  const markInvoiceAsPaid = async (payableId: string) => {
+    if (payableId) {
+      await payMutation.mutateAsync(
+        {
+          path: { payable_id: payableId },
+        },
+        {
+          onSuccess: (payable) => {
+            toast.success(
+              t(i18n)`Payable "${payable.document_id}" has been paid`
+            );
+          },
+        }
+      );
+    }
+  };
+
+  const handleCloseModal = async () => {
     setModalOpen(false);
     setIframeUrl(null);
+    await markInvoiceAsPaid(payableId);
   };
 
-  const handlePay = () => {
-    const paymentPageUrl = paymentLinkQuery?.data?.payment_page_url;
+  const handlePay = async () => {
+    if (!paymentLinkId) {
+      if (!returnUrl?.startsWith('https://')) {
+        toast.error(t(i18n)`Return URL must use HTTPS protocol`);
+        return;
+      }
+
+      const availablePaymentMethods = paymentMethods
+        ? paymentMethods.data.filter(
+            ({ status, direction }) =>
+              status === 'active' && direction === 'receive'
+          )
+        : [];
+
+      if (availablePaymentMethods.length === 0) {
+        toast.error(
+          t(
+            i18n
+          )`No active payment methods available. Please configure payment methods first.`
+        );
+        return;
+      }
+
+      try {
+        await createPaymentLinkMutation.mutateAsync({
+          recipient: {
+            id: counterpartId!,
+            type: 'counterpart',
+          },
+          object: {
+            id: payableId,
+            type: 'payable',
+          },
+          payment_methods: availablePaymentMethods.map((method) => method.type),
+          return_url: returnUrl,
+        });
+      } catch (error) {
+        console.error('Payment link creation error:', error);
```

Comment:
Did you intend to leave this error to be logged to the console?

Reply:
I did on purpose to test production possible issue on payment flow requests